### PR TITLE
✨amp-consent: Added support for allowing CMPs to add their own documentation and examples for amp-consent

### PR DESCRIPTION
**Pull request by torch2424**
Similar to `amp-analytics` and `amp-ads`. This lays the groundwork for 3p CMPs to integrate with `amp-consent`, and have a central example and documentation for their provider.

cc @zhouyx 

# Example

Screen Shot 2019-05-08 at 5 53 48 PM

### CODE REVIEW

**Comment by torch2424** [comment]
Also, just a heads up, once this is merged, we should update: https://github.com/ampproject/amphtml/pull/21693#issuecomment-481048450

File: examples/cmp-vendors.amp.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    <script async custom-element="amp-form" src="https://cdn.ampproject.org/v0/amp-form-0.1.js"></script>
+    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+      <script async src="https://cdn.ampproject.org/v0.js"></script>
+      <script async custom-element="amp-consent" src="https://cdn.ampproject.org/v0/amp-consent-0.1.js"></script>
+  </head>
+  <body>
+
+    <div class="filterBar">
```
Comment (zhouyx) [inline examples/cmp-vendors.amp.html]: How difficult it is to introduce a button to clear related localStorage? (I'm fine making this a non AMP page).

File: examples/cmp-vendors.amp.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          {
+            "consents": {},
+              "postPromptUI": "postPromptUI",
+              "clientConfig": {
+                "CMP_id": "test_id",
+                "other_info": "test_info"
+              }
+          }
+        </script>
+        <div id="ui1">
```
Comment (zhouyx) [inline examples/cmp-vendors.amp.html]: nit: 'ui1' and 'ui2' not used, can be removed

File: extensions/amp-consent/cmps/_ping_.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  {
+    "consents": {},
+      "postPromptUI": "postPromptUI",
+      "clientConfig": {
+        "CMP_id": "test_id",
+        "other_info": "test_info"
+      }
+  }
+  </script>
+  <div id="ui1">
```
Comment (zhouyx) [inline extensions/amp-consent/cmps/_ping_.md]: nit: can be removed

Comment (torch2424) [reply extensions/amp-consent/cmps/_ping_.md]: See above 😄

File: examples/cmp-vendors.amp.html (hunk 2 above)
Comment (torch2424) [reply examples/cmp-vendors.amp.html]: Yes it is? I wanted prompts so vendors can see their UI working:

Screen Shot 2019-05-08 at 5 53 48 PM

I think the confusion may have been from the experiment not being toggled? So I added a note as well. And updated the original PR screenshot 😄

File: examples/cmp-vendors.amp.html (hunk 1 above)
Comment (torch2424) [reply examples/cmp-vendors.amp.html]: Done! 😄

File: extensions/amp-consent/cmps/_ping_.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+<!---
```
[the remark this answers is not in the document]
Comment (torch2424) [reply extensions/amp-consent/cmps/_ping_.md]: Done! 😄

**Comment by torch2424** [comment]
@zhouyx This is good to go! 😄 Made all requested changes / replied to comments!

File: examples/cmp-vendors.amp.html (hunk 2 above)
Comment (zhouyx) [reply examples/cmp-vendors.amp.html]: I believe the UI is from the _ping_ iframe, where you specify using `'promptUISrc': '/test/manual/diy-consent.html',`

I don't think the `ui1` and `ui2` are used as they are not referred anywhere.

File: examples/cmp-vendors.amp.html (hunk 1 above)
Comment (zhouyx) [reply examples/cmp-vendors.amp.html]: Thank you!

File: examples/cmp-vendors.amp.html (hunk 2 above)
Comment (torch2424) [reply examples/cmp-vendors.amp.html]: Oh duh 😂 Thank you!

**Comment by torch2424** [comment]
@zhouyx This is good to go! 😄 Made all requested changes / replied to comments!

File: extensions/amp-consent/cmps/_ping_.md (hunk 3 above)
Comment (zhouyx) [reply extensions/amp-consent/cmps/_ping_.md]: ditto

**Comment by torch2424** [comment]
@MaximePlancke As a follow up to https://github.com/ampproject/amphtml/pull/21693#issuecomment-481048450 .

Since this is now merged, feel free to add your example / provide feedback on the process, in a separate PR. That way AMP users can see your platform when looking for CMPs 😄 

Thank you!

**Comment by MaximePlancke** [comment]
@torch2424 Thank you for your notification.

We will spend some time adding our CMP to your documentation most likely next week.

Thank you :)

**Comment by MaximePlancke** [comment]
@torch2424 The PR is ready (https://github.com/ampproject/amphtml/pull/22518) but we still need to finalize our documentation on our end. That is the reason why the job is failing because the link in the doc is not working yet. Once we finalize it I'll rerun the jobs and let you know.

Thank you

**Comment by torch2424** [comment]
@MaximePlancke Will move our discussion to the PR you linked.

No worries about the link, we are in no rush. 

Thank you very much for opening this follow up PR! 😄 👍

**Pull request merged**